internal/controlplane/handlers_profile.go

```diff
@@ -547,71 +549,63 @@ func (s *Server) PatchProfile(ctx context.Context, ppr *minderv1.PatchProfileReq
 		return nil, util.UserVisibleError(codes.InvalidArgument, "Malformed UUID")
 	}
 
-	// The UpdateProfile accepts Null values for the remediate and alert fields as "the server default". Until
-	// we fix that, we need to fetch the old profile to get the current values and extend the patch with those
-	oldProfile, err := s.store.GetProfileByID(ctx, db.GetProfileByIDParams{
-		ProjectID: entityCtx.Project.ID,
-		ID:        profileID,
-	})
+	oldProfilePb, err := getProfilePBFromDB(ctx, profileID, entityCtx, s.store)
 	if err != nil {
-		if errors.Is(err, sql.ErrNoRows) {
-			return nil, util.UserVisibleError(codes.NotFound, "profile not found")
-		}
 		return nil, status.Errorf(codes.Internal, "failed to get profile: %s", err)
 	}
 
-	// TEMPORARY HACK: Since we do not need to support this patch logic in the
```

Comment:
We still need this check.

Reply:
I can add a test but wouldn't the update code just error out later?